### PR TITLE
Add image scan docs

### PR DESCRIPTION
**Pull request by StrongMonkey**
https://github.com/rancher/fleet/issues/378

### CODE REVIEW

File: docs/imagescan.md [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+  # user must properly provide a secret that have write access to git repository
+  clientSecretName: secret 
+  # specify the commit pattern
+  imageScanCommit:
+    authorName: foo
+    authorEmail: foo@bar.com
+    messageTemplate: "update image"
+```
+
+Try pushing a new image tag, for example, `your.registry.com/repo/image:v1.0.3`. Wait for a while and there should be a new commit pushed into your git repository to change tag in deployment.yaml.
```
Comment (nickgerace) [inline docs/imagescan.md:104]: Perhaps we should use `<tag>` instead of `v1.0.3`

File: docs/imagescan.md [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+  template:
+    metadata:
+      labels:
+        app: redis
+        role: slave
+        tier: backend
+    spec:
+      containers:
+      - name: slave
+        image: your.registry.com/repo/image:v1.0.2 # {"$imagescan": "test-scan"}
```
Comment (nickgerace) [inline docs/imagescan.md:62]: Same here with the tag, perhaps: `<image>:<tag>`

File: docs/imagescan.md [hunk 3]
```diff
@@ -0,0 +1,105 @@
+# Image scan
+
+Image scan in fleet allows you to scan your image repository, fetch the desired image and update your git repository, 
+without the need to manually update your manifests.
+
+!!! hint "Experimental"
+    This feature is considered as experimental feature.
+    
+Go to fleet.yaml and add the following section.
```
Comment (nickgerace) [inline docs/imagescan.md:9]: nit: `fleet.yaml` instead of fleet.yaml

**Comment by philomory** [comment]
This documentation should _probably_ be more explicit about what a semver range is, with an example that isn't just `"*"` (it's worth noting that the SemVer spec doesn't include the word "range" at all, so you're left to guess whose notion of "range" we're using here). It's probably also worth explicitly calling out how prerelease versions are handled; for example, if you write `>= 1.0.0`, then no pre-release SemVer tag will be considered, but if you write `>= 1.0.0-0` then they will be. I had to go digging through the source code to find out that we're using the `Masterminds/semver` library, and then go look at their readme to see how things worked.

**Review by nickgerace — APPROVED** [review]

**Pull request merged**